nixos/modules/services/networking/bee.nix

```diff
@@ -0,0 +1,166 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.bee;
+in {
+  meta = {
+    # doc = ./bee.xml;
+    maintainers = with maintainers; [ attila-lendvai ];
+  };
+
+  ### interface
+
+  options = {
+    services.bee = {
+      enable = mkEnableOption "Start up Ethereum Swarm Bee";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.bee;
+        defaultText = "pkgs.bee";
+        example = "pkgs.bee-unstable";
+        description = "The package providing the bee binary for the service.";
+      };
+
+      ### NOTE let's call the nix variables the same as bee calls them
+
+      config = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/var/lib/bee/bee.yaml";
+        #description = "The configuration file path.";
+      };
+
+      api-addr = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = ''HTTP API listen address (default ":1633")'';
+      };
+
+      data-dir = mkOption {
+        type = types.path;
+        default = "/var/lib/bee/";
+        #description = "The data directory.";
+      };
+
+      extraConfig = mkOption {
+        type = types.lines;
+        default = "";
+        example = ''
+          password-file: /var/lib/bee/password
+          payment-threshold: 100000
+        '';
+        description = "Additional text to be appended at the end of the default <filename>bee.yaml</filename>. Note that it's ignored when a custom config file is specified.";
+      };
+    };
+  };
+
+  ### implementation
+
+#  config = mkIf cfg.enable {
+  config = {
+    systemd.services.bee =
```

Comment:
How about we use the [upstream unit](https://github.com/ethersphere/bee/blob/master/packaging/bee.service) instead, via `systemd.packages = [ cfg.package ];`?

Reply:
i still lack the necessary high-level perspective on Nix to answer this. and i don't really grasp what that line means, or how that affects the end results.

a concern of mine: i think, with weak confidence, that it's better to configure the executable by feeding it command line arguments based on the nix file, than emitting a generated config file and feeding that to the exe. if we use the upstream service, then it has a fixed, config file based `ExecStart`.

Reply:
note that all aspects of the bee configuration can be provided through command line arguments.